src/hooks/use-network-connection.ts

```diff
@@ -0,0 +1,18 @@
+import { useContext } from "react";
+import { NetworkConnection } from "andculturecode-javascript-core";
+import { NetworkConnectionContext } from "../contexts/network-connection-context";
+
+/**
+ * Hook that returns the current network connection information
+ */
+export const useNetworkConnection = (): NetworkConnection => {
+    const networkConnection = useContext(NetworkConnectionContext);
+
+    if (networkConnection === undefined) {
```

Comment:
Is this explicit `undefined` check intentional? I think we prefer loose `null` checks, generally speaking.